### PR TITLE
[eme] persistent-usage-record drm tests

### PR DESCRIPTION
**Pull request by mwatson2**
This PR is dependent on #3476.

### CODE REVIEW

File: encrypted-media/polyfill/edge-polyfill.js [hunk 1]
```diff
@@ -0,0 +1,193 @@
+(function() {
+
+    // This polyfill fixes the following problems with Edge browser
+    // (1) To retrieve a persisted usage record, you must use session type 'persistent-release-message' instead of 'ppersistent-usage-record'
```
Comment (jdsmith3000) [inline encrypted-media/polyfill/edge-polyfill.js]: nit; 'ppersistent-usage-record' typo.

**Comment by jdsmith3000** [comment]
I believe this is a relatively direct migration of the clearkey tests to multi-drm.  LGTM.

I have a dev here reviewing as well.  Please hold merge for any comments from him.

File: encrypted-media/drm-mp4-playback-persistent-usage-record-events.html [hunk 2]
```diff
@@ -0,0 +1,54 @@
+<!doctype html>
+<html>
+  <head>
+    <meta charset=utf-8>
+    <meta name="timeout" content="long">
+    <title>Encrypted Media Extensions: Successful Playback, persistent-usage-record session with Clear Key, mp4, event sequence</title>
```
Comment (jdsmith3000) [inline encrypted-media/drm-mp4-playback-persistent-usage-record-events.html]: nit: shouldn't include clear key

File: encrypted-media/drm-mp4-playback-persistent-usage-record.html [hunk 3]
```diff
@@ -0,0 +1,54 @@
+<!doctype html>
+<html>
+  <head>
+    <meta charset=utf-8>
+    <meta name="timeout" content="long">
+    <title>Encrypted Media Extensions: Successful Playback, persistent-usage-record session with Clear Key, mp4, event sequence</title>
```
Comment (jdsmith3000) [inline encrypted-media/drm-mp4-playback-persistent-usage-record.html]: nit: shouldn't include clear key

File: encrypted-media/drm-mp4-playback-retrieve-persistent-usage-record.html [hunk 4]
```diff
@@ -0,0 +1,55 @@
+<!doctype html>
+<html>
+  <head>
+    <meta charset=utf-8>
+    <meta name="timeout" content="long">
+    <title>Encrypted Media Extensions: Successful Playback, persistent-usage-record session with Clear Key, mp4, event sequence</title>
```
Comment (jdsmith3000) [inline encrypted-media/drm-mp4-playback-retrieve-persistent-usage-record.html]: nit: shouldn't include clear key

File: encrypted-media/scripts/playback-retrieve-persistent-usage-record.js [hunk 5]
```diff
@@ -83,7 +80,7 @@ function runTest(config, testname) {
 
                 }));
 
-                win.close();
+                //win.close();
```
Comment (jdsmith3000) [inline encrypted-media/scripts/playback-retrieve-persistent-usage-record.js]: Why this change?

File: encrypted-media/polyfill/edge-polyfill.js [hunk 6]
```diff
@@ -0,0 +1,193 @@
...
+                return self._surrogate.remove();
+            }).then( function() { return true; } );
+        }
+
+        // update()
+        //
+        // For a normal session, pass through, otherwise update the surrogate and close the proxy
+        MediaKeySession.prototype.update = function update( message )
+        {
+            if ( !this._removing ) return this._session.update( message );
```
Comment (jdsmith3000) [inline encrypted-media/polyfill/edge-polyfill.js]: nit:  style inconsistencies (for example, “if ( !this._removing )” here and and “if (!this._removing)” on line 106 below).

**Comment by jdsmith3000** [comment]
Comments from us are now complete.  All are small, but it would be good to clean them up as we go.

File: encrypted-media/scripts/playback-retrieve-persistent-usage-record.js (hunk 5 above)
Comment (mwatson2) [inline encrypted-media/scripts/playback-retrieve-persistent-usage-record.js]: This is a mistake. I commented out for debugging.

**Comment by jdsmith3000** [comment]
LGTM
